homeassistant/components/mqtt/discovery.py

```diff
@@ -351,10 +353,7 @@ def discovery_done(_: Any) -> None:
             0,
             job_type=HassJobType.Callback,
         )
-        for topic in (
-            f"{discovery_topic}/+/+/config",
-            f"{discovery_topic}/+/+/+/config",
-        )
+        for topic in discovery_topics
```

Comment:
```suggestion
        for topic in chain(
            (
                f"{discovery_topic}/{component}/+/config"
                for component in SUPPORTED_COMPONENTS
            ),
            (
                f"{discovery_topic}/{component}/+/+/config"
                for component in SUPPORTED_COMPONENTS
            ),
        )
```

Or, import SUPPORTED_COMPONENTS as COMPONENTS and do:

```
        for topic in chain(
            (f"{discovery_topic}/{component}/+/config" for component in COMPONENTS),
            (f"{discovery_topic}/{component}/+/+/config" for component in COMPONENTS),
        )
```